### DISCUSSION TITLE
Retrieve from the database and immediatelly save

### DISCUSSION THREAD

**Discussion by foxale** [kedro-org/kedro#1885] [Idea]
Let's say I want to load some table from the database using SQLTableDataSet and automatically save it as a CSV file somewhere in the `data/` space. I can see three potential workarounds:
1. Create a custom dataset inheriting after SQLTableDataSet and add saving logic myself
2. Create an empty Kedro node (with no python code) which would save the dataset as its `output`
3. Maybe play around with [transcoding datasets](https://kedro.readthedocs.io/en/stable/data/data_catalog.html#transcode-datasets)?

What's the Kedro-native way to achieve this?

**Comment by oscarvilla** [comment]
It have passed a lot of time.
I was working with something similar: taking data from BigQuery through a query and then saving the result in a .csv in bucket.
My approach was like you point in 2.
